soc/arm/atmel_sam/same70/soc.c

```diff
@@ -226,15 +226,12 @@ static ALWAYS_INLINE void clock_init(void)
 
 void z_arm_platform_init(void)
 {
-	if (IS_ENABLED(CONFIG_CACHE_MANAGEMENT) && IS_ENABLED(CONFIG_ICACHE)) {
+	if (IS_ENABLED(CONFIG_ICACHE_AUTO_ENABLE)) {
 		SCB_EnableICache();
-	} else {
-		SCB_DisableICache();
 	}
```

Comment:
This diff and other similar ones in the SAM folder remove `SCB_DisableICache` calls when `ICACHE_AUTO_ENABLE=n`. Were these disables redundant?

Reply:
Honestly, there seems to be a little bit of a chaos when to enable or disable the caches. The basic idea seems to be that CONFIG_INIT_ARCH_HW_AT_BOOT would take care to put the stuff back into its reset state. But at some points there was additionally mentioned that SystemInit could have enabled the caches already previously. If such a comment existed I kept the disable-calls. But for these occasions I assume somebody just wanted to be on the safe side. Which then again brings a little bit more confusion into the whole affair if the caches are now enabled or disabled. Therefore, I removed them.